### PR TITLE
fix(shares): Don't notify users that "Unshared" a calendar

### LINKED ISSUE #100: Support ignoring users who unshared from a calendar
https://github.com/nextcloud/server/pull/43117 brings a new share type which needs to be taken into account.

### PR DESCRIPTION
**Pull request by nickvergessen**
- [x] Requires https://github.com/nextcloud/server/pull/47494 to be backported
- [x] Bump server version to 30+

Fix #100

### CODE REVIEW

File: lib/EventListener.php [hunk 1]
```diff
@@ -222,18 +222,44 @@ protected function getUsersForShares(array $shares, string $owner): array {
...
+
+		if (!$groupAddedUsers) {
+			return $users;
+		}
+
+		/** @var \OCA\DAV\CalDAV\Sharing\Service $service */
+		$service = \OCP\Server::get(\OCA\DAV\CalDAV\Sharing\Service::class);
+		if (!method_exists($service, 'getUnshares')) {
+			return $users;
+		}
```
Comment (tcitworld) [inline lib/EventListener.php]: Is this needed if you intend to bump server version to 30+?

File: lib/EventListener.php [hunk 2]
```diff
@@ -222,18 +222,44 @@ protected function getUsersForShares(array $shares, string $owner): array {
...
 				}
 			}
 		}
 
-		return array_unique($users);
+		$users = array_unique($users);
+
+		if (!$groupAddedUsers) {
+			return $users;
+		}
```
Comment (tcitworld) [inline lib/EventListener.php:217]: I need to check how circle shares work. :eyes:

**Review by tcitworld — APPROVED** [review]

File: lib/EventListener.php (hunk 1 above)
Comment (nickvergessen) [reply lib/EventListener.php]: If I bump, then not.

File: lib/EventListener.php (hunk 2 above)
Comment (nickvergessen) [reply lib/EventListener.php:217]: I did not know that calendars can be shared with circles...

Comment (nickvergessen) [reply lib/EventListener.php:217]: Raised https://github.com/nextcloud/event_update_notification/issues/107

**Pull request merged**